### ISSUE TITLE
Change shredding to add atomic fields into `array_contains` as well, to support array-or-atomic `$eq`

### ISSUE THREAD

**Issue by tatu-at-datastax** [stargate/jsonapi#106]
Since mongo storage supports matching of either:

* Single atomic field value (like String `abc`)
* Array with any of elements being specific atomic value (like String `abc`)

Query for ```{ "field" : { "$eq" : "abc" } }``` would match both of following documents

```
{ "field" : "abc" }
```

```
{ "fields" : [ 123, "abc", false ] }
```

To support this filter we can make use of existing `array_contains` field on which array elements are stored, and additionally story values of atomic fields too. When this occurs we can change Equality filter to only look for match there and it will cover both cases without need for OR query on Cassandra.
Basically it will replace use of `query_text_values` (and 3 other) for equality queries (we may still need these for other filter operations).

See #109 for specification, details.
